victor/src/layout/mod.rs

```diff
@@ -160,14 +249,253 @@ fn same_formatting_context_block(
             inline: inline_size,
         },
     };
-    Fragment::Box(BoxFragment {
+    let fragment = Fragment::Box(BoxFragment {
         style: style.clone(),
         children,
         content_rect,
         padding,
         border,
         margin,
-    })
+    });
+    (fragment, absolutely_positioned_fragments)
+}
+
+fn absolutely_positioned_box(
+    absolute_containing_block: &ContainingBlock,
+    containing_block: &ContainingBlock,
+    tree_rank: usize,
+    style: &Arc<ComputedValues>,
+    contents: &BlockFormattingContext,
+) -> (Fragment, FlatVec<AbsolutelyPositionedFragment>) {
+    let cbis = absolute_containing_block.inline_size;
+    let padding = style.padding().map(|v| v.percentage_relative_to(cbis));
+    let border = style.border_width().map(|v| v.percentage_relative_to(cbis));
+    let pb = &padding + &border;
+    let box_size = style.box_size();
+
+    let computed_box_offsets = style
+        .box_offsets()
+        .map(|v| v.non_auto().map(|v| v.percentage_relative_to(cbis)));
+    let computed_margin = style
+        .margin()
+        .map(|v| v.non_auto().map(|v| v.percentage_relative_to(cbis)));
+
+    let computed_inline_size = box_size
+        .inline
+        .non_auto()
+        .map(|v| v.percentage_relative_to(cbis));
+    let computed_block_size = box_size.block.non_auto().and_then(|b| match b {
+        LengthOrPercentage::Length(l) => Some(l),
+        LengthOrPercentage::Percentage(p) => {
+            absolute_containing_block.block_size.map(|cbbs| cbbs * p)
```

Comment:
I think this call to `map` should be replaced with `unwrap`, and possibly calls to this functions delayed until the containing block’s block-direction size is known.

Reply:
This is completely gone.